### PR TITLE
DB-8082 Fix DB-7960 regression

### PR DESCRIPTION
**Pull request by msirek**
Fixes an incorrect result regression from DB-7960, caused by adding a CAST operator in a window function.
The root cause is poor support for expressions in aggregate functions of complex window function queries.
For details, please see [DB-8082](https://splicemachine.atlassian.net/browse/DB-8082?focusedCommentId=60300&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-60300).

### CODE REVIEW

**Comment by msirek** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/WindowResultSetNode.java [hunk 1]
```diff
@@ -736,6 +736,8 @@ private void addWindowFunctionColumns() throws StandardException {
             List<ValueNode> operands = windowFunctionNode.getOperands();
             int[] inputVIDs = new int[operands.size()];
             int i = 0;
+            ColumnPullupVisitor columnPuller =
+               new ColumnPullupVisitor(WindowFunctionNode.class, LOG, ((SingleChildResultSetNode) this.childResult).getChildResult(), rCtoCRfactory);
```
Comment (jaceklaskowski) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/WindowResultSetNode.java]: Can you move `((SingleChildResultSetNode) this.childResult).getChildResult()` to a separate local value?

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/WindowResultSetNode.java [hunk 2]
```diff
@@ -779,12 +781,12 @@ private void addWindowFunctionColumns() throws StandardException {
...
                                       ". Creating one for it in " + wdn.toString());
                     }
+                    exp.accept(columnPuller);
 
                     tmpRC = (ResultColumn) getNodeFactory().getNode(C_NodeTypes.RESULT_COLUMN,
-                                                                           "##" + windowFunctionNode.getAggregateName() +
-                                                                               "_Input_" + exp.getColumnName(),
-                                                                           exp,
-                                                                           getContextManager());
+                             "##" + windowFunctionNode.getAggregateName() + "_Input_" +
```
Comment (jaceklaskowski) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/WindowResultSetNode.java]: Can you move `"##" + windowFunctionNode.getAggregateName() + "_Input_" +...` to a separate local value, say `name` or `id`?

File: splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/AggregateAndArithmeticOverflowIT.java [hunk 3]
```diff
@@ -192,6 +192,38 @@ public static void createData(Connection conn, String schemaName) throws Excepti
 		spliceClassWatcher.executeUpdate(sqlText);
 		// 1M rows have been inserted
 
+        new TableCreator(conn)
+        .withCreate("create table MY_TABLE\n" +
```
Comment (jaceklaskowski) [inline splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/AggregateAndArithmeticOverflowIT.java]: indentation

File: splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/AggregateAndArithmeticOverflowIT.java [hunk 4]
```diff
@@ -192,6 +192,38 @@ public static void createData(Connection conn, String schemaName) throws Excepti
...
+                    "(\n" +
```
Comment (jaceklaskowski) [inline splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/AggregateAndArithmeticOverflowIT.java]: Just curious, are the newlines in the SQL required?

**Comment by msirek** [comment]
@jaceklaskowski Thank you for your comments.  The Splice code style and formatting standards are still evolving, and we will consider your recommendations moving forward.  For this specific JIRA we need to get it merged quickly to start the weekly build.

**Pull request merged**